frontend/packages/console-shared/src/components/actions/useActions.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  Action,
+  ActionProvider,
+  isActionProvider,
+  useResolvedExtensions,
+} from '@console/dynamic-plugin-sdk';
+
+const useActions = (contextId: string, scope: any): [Action[], boolean, any] => {
+  const [actionsMap, setActionsMap] = React.useState<{ [uid: string]: Action[] }>({});
+  const [loadError, setLoadError] = React.useState();
+  const stableProviderExtensions = React.useRef([]);
+
+  const actionProviderGuard = React.useCallback(
+    (e): e is ActionProvider => {
+      return isActionProvider(e) && e.properties.contextId === contextId;
+    },
+    [contextId],
+  );
+
+  const [providerExtensions, providerExtensionsResolved] = useResolvedExtensions<ActionProvider>(
+    actionProviderGuard,
+  );
+
+  React.useEffect(() => {
+    if (providerExtensionsResolved && stableProviderExtensions.current.length < 1) {
+      stableProviderExtensions.current = providerExtensions;
+    }
+    // Do not upodate the stable provider ref every time the extensions change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [providerExtensionsResolved]);
+
+  stableProviderExtensions.current.forEach((extension) => {
+    const [result, resolved, error] = extension.properties.provider(scope);
```

Comment:
Oh no... `useActions` may not be a viable option because of the lazy resolution of extensions...

Reply:
Yeah, I was trying to find a way to make it work. But it doesn't seem like it will work as expected. Will remove it from the commit.